core/src/main/java/org/elasticsearch/repositories/blobstore/ChecksumBlobStoreFormat.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.repositories.blobstore;
+
+import com.google.common.io.ByteStreams;
+import org.apache.lucene.codecs.CodecUtil;
+import org.apache.lucene.index.CorruptIndexException;
+import org.apache.lucene.index.IndexFormatTooNewException;
+import org.apache.lucene.index.IndexFormatTooOldException;
+import org.apache.lucene.store.OutputStreamIndexOutput;
+import org.elasticsearch.common.ParseFieldMatcher;
+import org.elasticsearch.common.blobstore.BlobContainer;
+import org.elasticsearch.common.bytes.BytesArray;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.compress.CompressorFactory;
+import org.elasticsearch.common.io.stream.BytesStreamOutput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.lucene.store.ByteArrayIndexInput;
+import org.elasticsearch.common.xcontent.*;
+import org.elasticsearch.gateway.CorruptStateException;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Locale;
+
+/**
+ * Snapshot metadata file format used in v2.0 and above
+ */
+public class ChecksumBlobStoreFormat<T extends ToXContent> extends BlobStoreFormat<T> {
+
+    public static final String TEMP_FILE_PREFIX = "pending-";
+
+    private static final XContentType DEFAULT_X_CONTENT_TYPE = XContentType.JSON;
+
+    // The format version
+    public static final int VERSION = 1;
+
+    private static final int BUFFER_SIZE = 4096;
+
+    protected final XContentType xContentType;
+
+    protected final boolean wrapper;
```

Comment:
Oh I see it below, makes sense, `wrapper` just sounds like a noun instead of a flag